addons/Commons/src/tts-utils.js

```diff
@@ -234,7 +287,168 @@
+            numberLength = n.toString().length;
+            var ordinalPartAsStr = this._getOrdinalPartOfNumber(n).toString();
+            var ordinalPartLength = ordinalPartAsStr.length;
+            var numberWithoutOrdinalPart = Math.floor(n / Math.pow(10, ordinalPartLength)) * Math.pow(10, ordinalPartLength);
+
+            var hasNonOrdinalPart = (ordinalPartLength !== numberLength);
+            if (!hasNonOrdinalPart)
+                return '';
+            var nAsText = this.numberToPolishNumber(numberWithoutOrdinalPart);
+
+            var ordinalPartSize = Math.floor((ordinalPartLength - 1) / 3);
+            var nonOrdinalPartWithoutOrdinalPartSize = Math.floor(n / Math.pow(10, (ordinalPartSize + 1) * 3)) * Math.pow(10, (ordinalPartSize + 1) * 3);
+
+            var ordinalPartIsABigNumber = (ordinalPartLength > 3);
+            var nonOrdinalPartContainsOrdinalPartSize = (nonOrdinalPartWithoutOrdinalPartSize !== numberWithoutOrdinalPart);
+
+            if (ordinalPartIsABigNumber && nonOrdinalPartContainsOrdinalPartSize)
+                nAsText = this._removeLastWord(nAsText);
+
+            return this._removeExtraSpaces(nAsText)
+        },
+
+        _getTextOrdinalPartOfNumber: function(n, gender) {
+            if (!gender) {
+                gender = this.GENDER.MASCULINE
+            }
+            var textUnits = ['', 'pierwszy', 'drugi', 'trzeci', 'czwarty', 'pi\u0105ty', 'sz\u00f3sty', 'si\u00f3dmy', '\u00f3smy', 'dziewi\u0105ty'];
+            var textTeens = ['', 'jedenasty', 'dwunasty', 'trzynasty', 'czternasty', 'pi\u0119tnasty', 'szesnasty', 'siedemnasty', 'osiemnasty', 'dziewi\u0119tnasty'];
+            var textTens = ['', 'dziesi\u0105ty', 'dwudziesty', 'trzydziesty', 'czterdziesty', 'pi\u0119\u0107dziesi\u0105ty', 'sze\u015b\u0107dziesi\u0105ty', 'siedemdziesi\u0105ty', 'osiemdziesi\u0105ty', 'dziewi\u0119\u0107dziesi\u0105ty'];
+            var textHundreds = ['', 'setny', 'dwusetny', 'trzysetny', 'czterysetny', 'pi\u0119\u0107setny', 'sze\u015b\u0107setny', 'siedemsetny', 'osiemsetny', 'dziewi\u0119\u0107setny'];
+
+            var textBigUnits = ['', 'jedno', 'dwu', 'trzy', 'cztero', 'pi\u0119cio', 'sze\u015bcio', 'siedmio', 'o\u015bmio', 'dziewi\u0119cio'];
+            var textBigTeens = ['', 'jedenasto', 'dwunasto', 'trzynasto', 'czternasto', 'pi\u0119tnasto', 'szesnasto', 'siedemnasto', 'osiemnasto', 'dziewi\u0119tnasto'];
+            var textBigTens = ['', 'dziesi\u0119cio', 'dwudziesto', 'trzydziesto', 'czterdziesto', 'pi\u0119\u0107dziesi\u0119cio', 'sze\u015b\u0107dziesi\u0119cio', 'siedemdziesi\u0119cio', 'osiemdziesi\u0119cio', 'dziewi\u0119\u0107dziesi\u0119cio'];
+            var textBigHundreds = ['', 'stu', 'dwustu', 'trzystu', 'czterystu', 'pi\u0119ciuset', 'sze\u015bciuset', 'siedmuset', 'o\u015bmiuset', 'dziewi\u0119ciuset'];
+            var sizes = ['', 'tysi\u0119czny', 'milionowy', 'miliardowy'];
+
+            textUnits.forEach(function(element, index, array) {

```

Comment:
spróbował bym wynieść ten kod:
 textTeens.forEach(function(element, index, array){
                array[index] = this._simpleChangePolishWordGender(element, gender);
}.bind(this));
do funkcji z parametrem array, która nadpisuje stan przekazanego array